### PR TITLE
WEO-74 New widget - Stepper Implemented with documentation

### PR DESCRIPTION
**Pull request by saadjavaidalvi-weoveri**
Steppers are implemented with three designs, including
- Icons Stepper
- Icon Text Stepper
- Counter Text Stepper

### CODE REVIEW

File: widgets/example/lib/main.dart
Comment (Awais3214) [inline widgets/example/lib/main.dart:1]: Found some unused commented code

**Review by Awais3214 — CHANGES_REQUESTED** [review]
Looks good apart from the commented unused code in main.dart and few lines of unused code in the other files as well.

**Review by Awais3214 — APPROVED** [review]
Looks good overall, found one or two lines of unwanted commented code in every file.

File: widgets/example/lib/main.dart
Comment (saadjavaidalvi-weoveri) [reply widgets/example/lib/main.dart:1]: Fixed

File: widgets/lib/steppers/steppers_model.dart [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    this.boxDecoration = const BoxDecoration(
+      color: Colors.white,
+      shape: BoxShape.circle,
+    ),
+  });
+}
+
+/// This model class is used to help define the sufix widget along with its states
+/// in all three states.
+class SufixWidgetStepper {
```
Comment (ibtihaajk7) [inline widgets/lib/steppers/steppers_model.dart]: Typo in the class name it should be SuffixWidgetStepper. (Suffix)

File: widgets/lib/steppers/steppers_model.dart [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  IconStepperItemStyle({
+    this.iconThemeData,
+    this.boxDecoration = const BoxDecoration(
+      color: Colors.white,
+      shape: BoxShape.circle,
+    ),
+  });
+}
+
+/// This model class is used to help define the sufix widget along with its states
```
Comment (ibtihaajk7) [inline widgets/lib/steppers/steppers_model.dart]: should be Suffix here as well

File: widgets/lib/steppers/steppers_model.dart [hunk 3]
```diff
@@ -0,0 +1,41 @@
+import 'package:flutter/material.dart';
+
+/// This style model is used to customize the stepper states that are `completed` `active` and `inactive`.
+/// You can change the styles for the items based on the states.
+class StepperStyle {
```
Comment (ibtihaajk7) [inline widgets/lib/steppers/steppers_model.dart:5]: The class is not used in the file, either consider removing it or moving it to separate file and import when needed.

**Review by ibtihaajk7 — CHANGES_REQUESTED** [review]
Left some comments to review, some are typos and one is a file structure change.
Please take note for future reviews @Awais3214.

File: widgets/lib/steppers/steppers_model.dart (hunk 1 above)
Comment (saadjavaidalvi-weoveri) [reply widgets/lib/steppers/steppers_model.dart]: Good point, fixed

File: widgets/lib/steppers/steppers_model.dart (hunk 2 above)
Comment (saadjavaidalvi-weoveri) [reply widgets/lib/steppers/steppers_model.dart]: fixed

File: widgets/lib/steppers/steppers_model.dart (hunk 3 above)
Comment (saadjavaidalvi-weoveri) [reply widgets/lib/steppers/steppers_model.dart:5]: This is used in the woi_steppers.dart file by WOIStepper widget for 3 variables.

Comment (ibtihaajk7) [reply widgets/lib/steppers/steppers_model.dart:5]: Okay I was just looking and standard placements. but looks good for now.

**Review by ibtihaajk7 — APPROVED** [review]

**Comment by Awais3214** [comment]
Ibtihaaj,

Thank you for mentioning me. 

Seeing how you review was helpful and I will definitely be vigilant in the future.

Just putting this out there, I did notice some of the typos but I thought I was just being a grammar nazi. Will not repeat this mistake again.

**Pull request merged**